Initial commit

rtarassov opened the pull request.

rtarassov committed:
Initial commit

mmetsa commented on src/main/java/com/solodev/clickergamebackend/ClickerGameBackEndApplication.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
```
This is not needed, this is the Main class, not a Controller

mmetsa commented on src/main/java/com/solodev/clickergamebackend/controller/PlayerController.java:
```diff
@@ -0,0 +1,29 @@
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@CrossOrigin(origins = "http://localhost:4200/")
```
CORS errors only happen when FE and BE are both hosted on your own computer, which isn't the case here, so probably should remove it or comment it, because we don't want to allow it in production

mmetsa commented on src/main/java/com/solodev/clickergamebackend/controller/PlayerController.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    @GetMapping
+    public PlayerModel findPlayerByToken(@RequestHeader(value = "Token") String tokenValue) {
```
Look into using ResponseEntity<PlayerModel>.
Wrapping the PlayerModel gives you the flexibility to add additional information to the response, for example, status codes.
Ex.
return ResponseEntity.ok().body(playerModel);
This gives the response a HTTP Status 200 "Success"

mmetsa commented on src/main/java/com/solodev/clickergamebackend/repository/PlayerRepository.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.List;
+
+@Repository
+public class PlayerRepository {
```
We don't want hardcoded stuff in production

mmetsa commented on src/main/resources/pages/hello-page.html:
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
What's this file used for?

mmetsa commented on src/test/java/com/solodev/clickergamebackend/ClickerGameBackEndApplicationTests.java:
```diff
@@ -0,0 +1,13 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
If you're not doing any integration testing right now, then it's probably best to remove this.
Later, you can add it back

rtarassov committed:
Trying to get a query from the database. Doesn't work yet.

rtarassov committed:
Wow, I can now find player details from the database using the player's unique token.

rtarassov committed:
Started using JDBC
Added /deliver endpoint to update totalEggsProduced and eggsInStorage columns in database.

rtarassov committed:
missed this file on my last commit somehow

rtarassov committed:
Upgraded spring boot to 3.2.0
Updated dependencies
Changed database from mysql to postgresql
Started using hibernate
Started using lombok annotation @Data and removed getters and setters.
